docker/gitpod/ubuntu-small-latest.Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM osgeo/gdal:ubuntu-small-latest
+RUN uname -a && cat /etc/os-release
+# RUN apt-get install -y software-properties-common
+# RUN add-apt-repository -y ppa:ubuntugis/ubuntugis-unstable
+    # results in "E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?"
+RUN apt-get update --fix-missing
+RUN apt-get install -y apt-utils
+
+### Gitpod user ###
+# https://community.gitpod.io/t/how-to-resolve-password-issue-in-sudo-mode-for-custom-image/2395/3
+# '-l': see https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user
+RUN apt-get install -y sudo
+RUN useradd -l -u 33333 -G sudo -md /home/gitpod -s /bin/bash -p gitpod gitpod \
+    # passwordless sudo for users in the 'sudo' group
+    && sed -i.bkp -e 's/%sudo\s\+ALL=(ALL\(:ALL\)\?)\s\+ALL/%sudo ALL=NOPASSWD:ALL/g' /etc/sudoers
+
+# full list dependencies according to
+# https://github.com/OSGeo/gdal/blob/6e6aff451dbcde450f051bff2f2e75ce6a4a3e6f/.github/workflows/cmake_builds.yml#L37
+RUN apt-get install -y bison libjpeg-dev libgif-dev liblzma-dev libzstd-dev libgeos-dev git \
```

Comment:
that's a bit fragile to have that duplication. couldn't that line be extracted in a shell script that would be run by the .yml and this docker file ?

Reply:
and why is that needed or that file and not osgeo-geo.Dockerfile ?

Reply:
I've no opinion on whether these are better in .yml/Dockerfile or shell script. I chose the former as that is what is used in the Gitpod docs (plus I'm a bash neophyte). I'll move into shell script.

As for why: because when I booted the image and tried to build and install gdal I'm told things are missing. The reason for trying both osgeo-gdal and ubuntu-small Dockerfiles is trying to optimize the boot time and network traffic. Small is what I personally prefer and my goal. 

(I'm continually astounded that these cloud computers are given away for free while at the same time I can't afford to build an equivalent machine for myself, and I'm not poor. I wish to keep as light a footprint as possible, so as not to pressure the upstream company, who have staff and bills to pay, to need to move this behind a subscription curtain.)

Reply:
@rouault thinking on this more, should the dependency lists in places like https://github.com/OSGeo/gdal/blob/master/.github/workflows/cmake_builds.yml#L37 also be in shell script? Come to think of it, apt probably has a concept similar to `pip install -r requirements.txt` too (?)

Reply:
...apparently apt doesn't have this baked in, however there I did find https://github.com/seatgeek/bash-aptfile by way of https://stackoverflow.com/questions/10121210/requirements-file-for-apt-get-similar-to-pip which also has the simple `xargs apt-get install <reqs.txt` incantation. What place in the source code tree should such a requirements file live?

Reply:
>  thinking on this more, should the dependency lists in places like https://github.com/OSGeo/gdal/blob/master/.github/workflows/cmake_builds.yml#L37 also be in shell script?

That's an option. There are different ways of structuring CI scripts.

> Come to think of it, apt probably has a concept similar to pip install -r requirements.txt too (?)

https://unix.stackexchange.com/a/212214 but note that the name of packages can change slightly between ubuntu versions, contrary to pip

Reply:
> What place in the source code tree should such a requirements file live?

I'm not sure. Maybe requirements/ubuntu-20.04-dev-package-list-full.txt ?

Reply:
I've abandoned the approach of using custom dockerfiles and am focussing on just using the latest osgeo published  image. There's just too much overlap with the Github workflows and friends and I haven't made appreciable headway in abstracting out the common bits.